sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -32,6 +32,16 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
   import org.apache.spark.sql.connector.catalog.CatalogV2Util._
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+    case unresolved @ UnresolvedRelation(nameParts) =>
+      nameParts match {
+        case AsTemporaryViewIdentifier(i) if catalogManager.v1SessionCatalog.isTemporaryTable(i) =>
```

Comment:
This was added to convert to v2 unresolved relation. Matching the view identifier ensures the same behavior as `lookupV2RelationAndCatalog`.